src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTestBase.cs

```diff
@@ -323,5 +449,87 @@ public async Task<string> GetRepositoryPolicies(string repoUri, string branchNam
         {
             return await RunDarcAsync("get-repository-policies", "--all", "--repo", repoUri, "--branch", branchName);
         }
+
+        public void CheckGitDirectory(string path, string gitDirFolder)
+        {
+            string[] expectedFolders = { "hooks", "info", "logs", "objects", "refs" };
+            foreach (string folder in expectedFolders)
+            {
+                string folderPath = Path.Combine(gitDirFolder, folder);
+                Assert.IsTrue(Directory.Exists(folderPath), $"{path} does not appear to be a valid .gitdir: missing folder '{folder}'");
+            }
+
+            string[] expectedFiles = { "config", "description", "FETCH_HEAD", "HEAD", "index" };
+            foreach (string file in expectedFiles)
+            {
+                string filePath = Path.Combine(gitDirFolder, file);
+                Assert.IsTrue(File.Exists(filePath), $"{path} does not appear to be a valid .gitdir: missing file '{file}'");
+            }
+        }
+
+        public void VerifyClonedDirectories(Dictionary<string, string> repos, List<string> masterRepos, List<string> gitDirs, string gitDirFolder, string reposFolder)
+        {
+            foreach (string name in repos.Keys)
+            {
+                string path = Path.Combine(reposFolder, name);
+                bool checkDir = Directory.Exists(path);
+                Assert.IsTrue(Directory.Exists(path), $"Expected cloned repo '{name}' but not found at '{path}'");
+
+                string versionDetailsPath = Path.Combine(path, "eng", "Version.Details.xml");
+                if (File.Exists(versionDetailsPath))
+                {
+                    using (FileStream stream = File.OpenRead(versionDetailsPath))
+                    {
+                        SHA256 Sha256 = SHA256.Create();
+                        byte[] hash = Sha256.ComputeHash(stream);
+                        string hashString = BitConverter.ToString(hash);
+                        Assert.AreEqual(hashString, repos[name], $"Expected {versionDetailsPath} to have hash '{repos[name]}', actual hash '{hashString}'");
```

Comment:
We are... hashing the Versions file? That's going to make the failure really hard to debug. Though I don't see any calls to this method...

Reply:
This is what's done in the repo cloning powershell test, which is still in progress. I'm absolutely open to better ideas for any of the cloning stuff, it's fairly murky in a lot of places.